pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -48,8 +48,8 @@ import (
 
 const (
 	randomMacGenerationAttempts = 10
-	qemuUID                     = "107"
-	qemuGID                     = "107"
+	qemuUID                     = "0"
+	qemuGID                     = "0"
```

Comment:
hmm... I think we don't want to make things worse and let qemu run as root, right?
I still think that a proper way to remove CAP_NET_ADMIN is by moving libvirt to run as non-root and use user-session with qemu.

@davidvossel @rmohr what do you think?

Reply:
Oh I see why you get that impression; this doesn't make qemu run as root; this only makes the tap device owner / group be 0 (thus matching the libivrt user).

I.e. it has to be the same as libvirt (since libvirt `ioctl`s the tap) and passes the opened file descriptor to qemu. 

QEMU will still be run by qemu.

Long term, you're right: libvirt should run as non-root, and we should create the tap with whatever user UID / GID libvirt runs under. But right now, we can remove the capability just by matching the tap creator / consumer. I just mistakenly assumed the consumer was qemu. It's libvirt.

Reply:
if libvirt already runs as root, why does it matter what the tap device runs as? libvirt would have permission to ioctl regardless of the user, right? Maybe there are some details i'm missing here?

Reply:
Yup, but as usually, the details are practically all that matter.

CAP_NET_ADMIN is only required if the users of the tap creator / whoever `ioctl(..., TUNSETIFF,...)` it do not match, as per [0] & [1].

[0] - https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/tun.c?id=4ef8451b332662d004df269d4cdeb7d9f31419b5#n574
[1] - https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/tun.c?id=4ef8451b332662d004df269d4cdeb7d9f31419b5#n2678

Reply:
> Oh I see why you get that impression; this doesn't make qemu run as root; this only makes the tap device owner / group be 0 (thus matching the libivrt user).
> 
Yep, it jumped on top of my list and I've reacted too fast.


> I.e. it has to be the same as libvirt (since libvirt `ioctl`s the tap) and passes the opened file descriptor to qemu.
>
 
> QEMU will still be run by qemu.
> 
> Long term, you're right: libvirt should run as non-root, and we should create the tap with whatever user UID / GID libvirt runs under. But right now, we can remove the capability just by matching the tap creator / consumer. I just mistakenly assumed the consumer was qemu. It's libvirt.